lib/routes/routeMetadata.js

```diff
@@ -0,0 +1,175 @@
+        'content-type': 'application/json',
+        'content-length': Buffer.byteLength(body),
+    };
+    response.writeHead(200, httpHeaders);
+    response.end(body, 'utf8', callback);
+}
+
+/*
+- getting list of buckets for each raft session -
+GET /_/metadata/listbuckets/<raft session id>
+- getting list of objects for each bucket -
+GET /_/metadata/getbucket/<bucket name>
+- getting metadata for bucket -
+GET /_/metadata/listobjects/<bucket name>
+- getting metadata for object -
+GET /_/metadata/getobject/<bucket name>/<object key>
+*/
+
+function getRaftBuckets(request, response, raftId, objectKey, log, callback) {
+    return metadata.getRaftBuckets(raftId, log, (err, res) => {
+        if (err) {
+            // TODO: figure out correct error to respond with
+            return callback(err);
+        }
+        return _respond(response, res, log, callback);
+    });
+}
+
+function getBucketMetadata(request, response, bucketName, objectKey, log,
+    callback) {
+    return metadata.getBucketAttributes(bucketName, log, (err, res) => {
+        if (err) {
+            // TODO: figure out correct error to respond with
+            return callback(err);
+        }
+        return _respond(response, res, log, callback);
+    });
+}
+
+function getObjectList(request, response, bucketName, objectKey, log,
+    callback) {
+    return metadata.listObject(bucketName, { listingType: 'Delimiter' }, log,
+    (err, res) => {
+        if (err) {
+            return callback(err);
+        }
+        return _respond(response, res, log, callback);
+    });
+}
+
+function getObjectMetadata(request, response, bucketName, objectKey,
+log, callback) {
+    return metadata.getObject(bucketName, objectKey, { }, log, (err, res) => {
+        if (err) {
+            return callback(err);
+        }
+        return _respond(response, res, log, callback);
+    });
+}
+
+const metadataRoutes = {
+    GET: {
+        listbuckets: getRaftBuckets,
+        listobjects: getObjectList,
+        getbucket: getBucketMetadata,
+        getobject: getObjectMetadata,
+    },
+};
+
+function routeMetadata(clientIP, request, response, log) {
+    log.debug('routing request', { method: 'routeBackbeat' });
+    normalizeMetadataRequest(request);
+    const useMultipleBackend = request.resourceType === 'multiplebackenddata';
+    const invalidRequest = ((!request.resourceType || !request.bucketName) ||
+        (request.resourceType === 'getobject' && !request.objectKey) ||
+        (!request.query.operation && useMultipleBackend));
+    if (invalidRequest) {
+        log.debug('invalid request', {
+            method: request.method, bucketName: request.bucketName,
+            objectKey: request.objectKey, resourceType: request.resourceType,
+            query: request.query,
+        });
+        return responseJSONBody(errors.MethodNotAllowed, null, response, log);
+    }
+    const requestContexts = prepareRequestContexts('objectReplicate', request);
```

Comment:
Feels weird to use `objectReplicate` as a context here, which is also technical debt for backbeat (as backbeat does more than just replicating an object). But there may be implications with Vault too, I don't remember exactly, so it's possible that it is quicker to stick with this context.

If you have time to find out, can you add a comment?